aptos-move/mvhashmap/src/versioned_aggregators.rs

```diff
@@ -423,50 +425,197 @@ impl VersionedAggregators {
+                        ));
+                    }
+                },
+                AggregatorEntry::Apply(SnapshotDerived { .. }) => {
+                    // Because Derived values can depend on the current value, we need to compute other values before it.
+                    derived_ids.push(id);
+                    None
+                },
+                AggregatorEntry::Estimate(_) => {
+                    return Err(CommitError::CodeInvariantError(
+                        "Cannot commit an estimate".to_string(),
+                    ))
+                },
+            };
+
+            if let Some(new_entry) = new_entry {
+                versioned_value.insert(idx_to_commit, new_entry);
+            }
+        }
+
+        for id in derived_ids {
+            let mut versioned_value = self
+                .values
+                .get_mut(&id)
+                .expect("Value in commit needs to be in the HashMap");
+            let entry_to_commit = versioned_value
+                .versioned_map
+                .get(&idx_to_commit)
+                .expect("Value in commit at that transaction version needs to be in the HashMap");
+            let new_entry = match &**entry_to_commit {
+                AggregatorEntry::Apply(SnapshotDerived {
+                    base_snapshot,
+                    formula,
+                }) => {
+                    let prev_value = self.values
+                        .get_mut(base_snapshot)
+                        .ok_or_else(|| CommitError::CodeInvariantError("Cannot find base_aggregator for Apply(SnapshotDelta) during commit".to_string()))?
+                        // Read values committed in this commit
+                        .read_latest_committed_value(idx_to_commit + 1)
+                        .map_err(|e| CommitError::CodeInvariantError(format!("Cannot read latest committed value for base aggregator for ApplySnapshotDelta) during commit: {:?}", e)))?;
+
+                    if let AggregatorValue::Snapshot(base) = prev_value {
+                        let new_value = formula.apply(base);
+                        AggregatorEntry::Value(
+                            AggregatorValue::Derived(new_value),
+                            Some(SnapshotDerived {
+                                base_snapshot: *base_snapshot,
+                                formula: formula.clone(),
+                            }),
+                        )
+                    } else {
+                        return Err(CommitError::CodeInvariantError(
+                            "Cannot apply delta to non-AggregatorValue::Aggregator".to_string(),
+                        ));
+                    }
+                },
+                _ => unreachable!("We've only added derived values into derived_ids"),
+            };
+
+            versioned_value.insert(idx_to_commit, new_entry);
+        }
+
         self.next_idx_to_commit
```

Comment:
fetch_add(1) here is sufficient, since it will ensure they all atomically increment by 1, and the check in the beginning of the method will catch the same code invariant errors